test/exercises/diffusion/heat_2d_dist_exchanges.chpl

```diff
@@ -0,0 +1,141 @@
+/*
+  A distributed 2D finite-difference heat/diffusion equation solver
+
+  Computation is executed over a 2D distributed array.
+  The array distribution is managed by the `Block` distribution.
+  Tasks are spawned manually with a `coforall` loop and synchronization
+  is done manually using a `barrier`. Halo regions are shared across
+  locales manually via direct assignment between locally owned arrays.
+*/
```

Comment:
Oh wow, another version!?  Can you update the OP to mention the versions you're adding here and what their differences in approach are, and maybe consider adding a README to this directory that does the same.  I need a lunch break before reviewing this one, so am going to push my review for now and come back to it later.  Note that this version doesn't have .perf* files and isn't in the .graphs file—is that intentional?